CONTRIBUTING.md

```diff
@@ -2,15 +2,216 @@ Guidelines for Contributing
+    all concerns, don't hesitate to ping your reviewers
+-   Ping [@9il](https://github.com/9il) for anything math related
+-   See [rebasing](https://wiki.dlang.org/Starting_as_a_Contributor#Rebasing) if you
+    need to sync your fork with master
+
+Naming guidelines
+-----------------
+
+### General advices
+
+-   Phobos uses a distinct [naming
+    scheme](https://wiki.dlang.org/Naming_conventions)
+-   Use a Capitalized short (50 chars or less) Git commit summary
+-   If necessary, wrap your Git explanation paragraph [to 72
+    chars](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)
+-   Do you use clear variable names?
+-   Does your PR have a concise and informative title?
+-   Are your Git messages clear and informative (otherwise use `git`
+    `commit` `--amend` for a single commit or `git` `rebase` for
+    multiple commits)?
+-   Avoid having many commits per PR (usually one commit per PR) - see
+    [squashing](https://wiki.dlang.org/Starting_as_a_Contributor#Someone_asked_me_to_squash_my_commits.2C_what_does_that_mean.3F) to combine commits
+
+### Issues
+
+-   Bug fix PRs should have a separate commit for each bugzilla issue
+    that it fixes
+-   The message of the commit actually fixes the bug should start with
+    "Fix issue X" where X is the number of the corresponding
+    [bugzilla](https://issues.dlang.org) issue. For example:
+
+```
++ Refactor the code assist with the bug fix
++ Fix issue 17000 - map doesn't accept some range types
++ Fix issue 17001 - map doesn't work with some other range type
+```
+
+-   Unrelated fixes, refactorings or additions should preferably go
+    separate pull request(s) (tightly related fixes are okay in the
+    same PR)
+
+Documentation
+-------------
+
+### Documentation style
+
+-   The documentation is written in
+    [ddoc](http://dlang.org/spec/ddoc.html)
+-   Use complete English sentences with correct syntax, grammar, and
+    punctuation
+-   Use precise technical terms. Avoids colloquialisms
+-   Honestly describe limitations and dangers
+-   Add `Params`, `Returns` and if needed `See_Also` sections
+-   Provide at least one ddoced unittest for a new method
+-   Have a brief one sentence summary for at the beginning of a method
+-   Give a synopsis at the beginning of a module accompanied by an
+    example
+
+### Documentation checks
+
+-   Read your documentation text aloud - is it understandable?
+-   Check the output of CyberShadows's DAutoTest of the documentation
+    (or run
+    [it](https://github.com/D-Programming-Language/dlang.org) yourself)
+-   All user impacting changes should have a corresponding
+    [changelog](https://github.com/D-Programming-Language/phobos/blob/master/changelog.dd)
+    entry
+-   Did you add your method to the summary booktable or cheatsheat of a
+    module? (might be in `package.d`)
+
+### Documentation tricks
+
+-   Use backticks `` `code` `` instead of `$(D` `foo)`
```

Comment:
Hmm, but I saw a couple of nitpicks to use the "new" backticks style. Isn't this the the style (?)